lib/improver/utilities/cube_checker.py

```diff
@@ -52,3 +56,125 @@ def check_for_x_and_y_axes(cube):
             msg = ("The cube does not contain the expected {}"
                    "coordinates.".format(axis))
             raise ValueError(msg)
+
+
+def check_cube_coordinates(cube, new_cube, exception_coordinates=None):
+    """Find and promote to dimension coordinates any scalar coordinates in
+    new_cube that were originally dimension coordinates in the progenitor
+    cube. If coordinate is in new_cube that is not in the old cube, keep
+    coordinate in its current position.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube
+        The input cube that will be checked to identify the preferred
+        coordinate order for the output cube.
+    new_cube : iris.cube.Cube
+        The cube that must be checked and adjusted using the coordinate order
+        from the original cube.
+    exception_coordinates : List of strings or None
+        The names of the coordinates that are permitted to be within the
+        new_cube but are not available within the original cube.
+
+    Returns
+    -------
+    new_cube : iris.cube.Cube
+        Modified cube with relevant scalar coordinates promoted to
+        dimension coordinates with the dimension coordinates re-ordered,
+        as best as can be done based on the original cube.
+
+    Raises
+    ------
+    CoordinateNotFoundError raised if the final dimension
```

Comment:
I think this may need to be formatted differently, See Caroline's formatting in WeightedBlend.process